<a href="https://colab.research.google.com/github/rach-zhang/data_analytics_2/blob/main/Salaries.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Wranging Assignment: Salaries

# Environment Setup

In [ ]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [ ]:
df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/master/Salaries.csv')

# Data Exploration

In [ ]:
type(df)

pandas.core.frame.DataFrame

In [ ]:
df.shape

(397, 7)

In [ ]:
df.head(10)

Unnamed: 0       rank discipline  yrs.since.phd  yrs.service     sex  salary
0           1       Prof          B             19           18    Male  139750
1           2       Prof          B             20           16    Male  173200
2           3   AsstProf          B              4            3    Male   79750
3           4       Prof          B             45           39    Male  115000
4           5       Prof          B             40           41    Male  141500
5           6  AssocProf          B              6            6    Male   97000
6           7       Prof          B             30           23    Male  175000
7           8       Prof          B             45           45    Male  147765
8           9       Prof          B             21           20    Male  119250
9          10       Prof          B             18           18  Female  129000

In [ ]:
df.tail()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
392         393      Prof          A             33           30  Male  103106
393         394      Prof          A             31           19  Male  150564
394         395      Prof          A             42           25  Male  101738
395         396      Prof          A             25           15  Male   95329
396         397  AsstProf          A              8            4  Male   81035

In [ ]:
df.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  397.000000     397.000000   397.000000     397.000000
mean   199.000000      22.314861    17.614610  113706.458438
std    114.748275      12.887003    13.006024   30289.038695
min      1.000000       1.000000     0.000000   57800.000000
25%    100.000000      12.000000     7.000000   91000.000000
50%    199.000000      21.000000    16.000000  107300.000000
75%    298.000000      32.000000    27.000000  134185.000000
max    397.000000      56.000000    60.000000  231545.000000

# Mean/Median of all salaries

In [ ]:
df['salary'].mean()

113706.45843828715

In [ ]:
df['salary'].median()

107300.0

# Median salary between gender and/or rank

Making a new dataframe that only has the columns 'sex' and 'salary'

In [ ]:
sex_salary = df[['sex','salary']]
print(sex_salary)

      sex  salary
0    Male  139750
1    Male  173200
2    Male   79750
3    Male  115000
4    Male  141500
..    ...     ...
392  Male  103106
393  Male  150564
394  Male  101738
395  Male   95329
396  Male   81035

[397 rows x 2 columns]


In [ ]:
male_salary = sex_salary[sex_salary['sex']=="Male"]
print(male_salary)

      sex  salary
0    Male  139750
1    Male  173200
2    Male   79750
3    Male  115000
4    Male  141500
..    ...     ...
392  Male  103106
393  Male  150564
394  Male  101738
395  Male   95329
396  Male   81035

[358 rows x 2 columns]


108043.0

In [ ]:
#median male salary
male_salary['salary'].median()

108043.0

In [ ]:
female_salary = sex_salary[sex_salary['sex']=="Female"]
print(female_salary.head())

       sex  salary
9   Female  129000
19  Female  137000
24  Female   74830
34  Female   80225
35  Female   77000


In [ ]:
female_salary.describe()

salary
count      39.000000
mean   101002.410256
std     25952.127317
min     62884.000000
25%     77250.000000
50%    103750.000000
75%    117002.500000
max    161101.000000

In [ ]:
#median female salary
female_salary['salary'].median()

103750.0

In [ ]:
#difference between mean salaries of male and female
male_salary['salary'].mean() - female_salary['salary'].mean()

14088.008738003162

# Data Wrangling

People who have 20 years of experience

In [ ]:
over_20_exp = df[df['yrs.service']>=20]
print(over_20_exp)

     Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
3             4  Prof          B             45           39  Male  115000
4             5  Prof          B             40           41  Male  141500
6             7  Prof          B             30           23  Male  175000
7             8  Prof          B             45           45  Male  147765
8             9  Prof          B             21           20  Male  119250
..          ...   ...        ...            ...          ...   ...     ...
384         385  Prof          A             27           21  Male  125192
386         387  Prof          A             29           27  Male  139219
388         389  Prof          A             38           36  Male  119450
392         393  Prof          A             33           30  Male  103106
394         395  Prof          A             42           25  Male  101738

[157 rows x 7 columns]


In [ ]:
print(over_20_exp['salary'].count())
print(over_20_exp['salary'].mean())

157
122520.65605095541


Comparing the salaries of people who have 20+ years of experience vs less than 20 years of experience

In [ ]:
under_20_exp = df[df['yrs.service']<20]
print(under_20_exp)

     Unnamed: 0       rank discipline  ...  yrs.service     sex  salary
0             1       Prof          B  ...           18    Male  139750
1             2       Prof          B  ...           16    Male  173200
2             3   AsstProf          B  ...            3    Male   79750
5             6  AssocProf          B  ...            6    Male   97000
9            10       Prof          B  ...           18  Female  129000
..          ...        ...        ...  ...          ...     ...     ...
390         391       Prof          A  ...           19    Male  166605
391         392       Prof          A  ...           19    Male  151292
393         394       Prof          A  ...           19    Male  150564
395         396       Prof          A  ...           15    Male   95329
396         397   AsstProf          A  ...            4    Male   81035

[240 rows x 7 columns]


In [ ]:
print(under_20_exp['salary'].count())
print(under_20_exp['salary'].mean())

240
107940.50416666667


In [ ]:
over_20_exp['salary'].mean() - under_20_exp['salary'].mean()

14580.151884288745

People who have salary of greater than or equal to 200,000

In [ ]:
over_200k = df[df['salary']>=200000]
print(over_200k)

     Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
43           44  Prof          B             38           38  Male  231545
249         250  Prof          A             29            7  Male  204000
364         365  Prof          A             43           43  Male  205500


# Findings
In this college, there are many more male professors than female professors. There are 358 male professors and only 39 female professors. On average, female professors earn 14000 less than their male counterparts.

This college has 157 professors with 20 or more years of experience and 240 professors with fewer than 20 years of experience. Professors with 20 or more years of experience have a greater salary but only by 14,600. Three out of 396 professors have salaries greater than 200,000.